modules/extended_tags.py

```diff
@@ -0,0 +1,58 @@
+"""
+    modules.extended_tags
+    ~~~~~~~~~~~~~~~~~~~~
+
+    Gives more power to tagging.
+"""
+
+from collections import Counter
+from collections import defaultdict
+
+
+def relatedness(taglist):
+    related = defaultdict(lambda: defaultdict(int))
+    for tag_set in taglist:
+        for tag in tag_set:
+            related_tags = tag_set.symmetric_difference({tag})
+            for other_tag in related_tags:
+                related[tag][other_tag] += 1
+    return related
+
+
+def make_get_related_tags(builder):
+    tags = get_all_tags(builder)
+
+    def get_related_tags(taglist):
+        rels = relatedness(tags)
```

Comment:
This can go outside the closure so it doesn't need to be recomputed each time get_related_tags is called